test/current_attributes_test.rb

```diff
@@ -94,6 +101,15 @@ class OtherCurrent < ActiveSupport::CurrentAttributes
     end
   end
 
+  it "persists after perform_inline" do
+    Sidekiq::CurrentAttributes.persist("Myapp::Current", @config)
+    with_context("Myapp::Current", :user_id, 16) do
+      assert_equal 16, Myapp::Current.user_id
+      CurrentAttributesJob.perform_inline
+      assert_equal 16, Myapp::Current.user_id
```

Comment:
Is this really the semantic we want? We mutated a global variable but didn’t reset it. It feels wrong.

Reply:
This code is really complex in its side effects and details so I’m struggling to understand the best approach. The value was already set when we called perform_inline so resetting it would be wrong too. I think this is better.